src/Forms/JoinSubtitles.cs

```diff
@@ -69,162 +70,137 @@ private void listViewParts_DragEnter(object sender, DragEventArgs e)
-                        }
-                    }
-
-                    if (format == null)
-                    {
-                        for (int j = k; j < _fileNamesToJoin.Count; j++)
-                        {
-                            _fileNamesToJoin.RemoveAt(j);
-                        }
-
-                        MessageBox.Show("Unkown subtitle format: " + fileName);
+                        CancelMergeOperation(i, fileName);
                         return;
                     }
-                    if (sub.Header != null)
+
+                    // use last subtitle header if available
+                    if (!string.IsNullOrEmpty(sub.Header))
                     {
                         header = sub.Header;
                     }
 
-                    if (lastFormat == null || lastFormat.FriendlyName == format.FriendlyName)
+                    // set last-format if not set yet
+                    if (lastFormat == null)
                     {
                         lastFormat = format;
                     }
                     else
                     {
-                        lastFormat = new SubRip(); // default subtitle format
+                        // uset default format as last if previous and current subtitle doesn't match formats
+                        if (!lastFormat.FriendlyName.Equals(format.FriendlyName))
+                        {
+                            lastFormat = OutputFormat;
+                        }
                     }
 
                     subtitles.Add(sub);
                 }
                 catch (Exception exception)
                 {
-                    for (int j = k; j < _fileNamesToJoin.Count; j++)
-                    {
-                        _fileNamesToJoin.RemoveAt(j);
-                    }
-
+                    CancelMergeOperation(i, fileName);
                     MessageBox.Show(exception.Message);
                     return;
                 }
             }
-            JoinedFormat = lastFormat;
 
-            for (int outer = 0; outer < subtitles.Count; outer++)
-            {
-                for (int inner = 1; inner < subtitles.Count; inner++)
-                {
-                    var a = subtitles[inner - 1];
-                    var b = subtitles[inner];
-                    if (a.Paragraphs.Count > 0 && b.Paragraphs.Count > 0 && a.Paragraphs[0].StartTime.TotalMilliseconds > b.Paragraphs[0].StartTime.TotalMilliseconds)
-                    {
-                        string t1 = _fileNamesToJoin[inner - 1];
-                        _fileNamesToJoin[inner - 1] = _fileNamesToJoin[inner];
-                        _fileNamesToJoin[inner] = t1;
+            OutputFormat = lastFormat;
 
-                        var t2 = subtitles[inner - 1];
-                        subtitles[inner - 1] = subtitles[inner];
-                        subtitles[inner] = t2;
-                    }
-                }
-            }

```

Comment:
replaced w/ linq with is more readable and which also made it easier to generate view